Sources/SwiftDocC/Infrastructure/Symbol Graph/SymbolGraphRelationshipsBuilder.swift

```diff
@@ -250,6 +250,24 @@ struct SymbolGraphRelationshipsBuilder {
         requiredSymbol.isRequired = true
     }
     
+    /// Adds a relationship from a type member to a protocol requirement.
+    /// - Parameters:
+    ///   - edge: A symbol graph relationship with a source and a target.
+    ///   - bundle: A documentation bundle.
+    ///   - symbolIndex: A symbol lookup map by precise identifier.
+    ///   - engine: A diagnostic collecting engine.
+    static func addOptionalRequirementRelationship(edge: SymbolGraph.Relationship, in bundle: DocumentationBundle, symbolIndex: inout [String: DocumentationNode], engine: DiagnosticEngine) {
+        // Resolve source symbol
+        guard let requiredNode = symbolIndex[edge.source],
```

Comment:
The implementation here is largely the same as `addRequirementRelationship(…)` above. Should we try and refactor these?